src/lib/Display.ts

```diff
@@ -0,0 +1,138 @@
+	 * Output message to stdout (non-blocking) only if the "--json" flag is not present.
+	 */
+	displayInfo(msg: string, ...args: any[]): void;
+
+	/**
+	 * Output message to stdout (non-blocking) only if the "--verbose" flag is present and the "--json" flag is not.
+	 */
+	displayVerboseInfo(msg: string): void;
+
+	/**
+	 * Display confirmation prompt. Times out and throws after 10s.
+	 */
+	displayConfirmationPrompt(msg: string): Promise<boolean>;
+
+	/**
+	 * Output styled header to stdout only if the "--json" flag is not present.
+	 */
+	displayStyledHeader(headerText: string): void;
+
+	/**
+	 * Output table to stdout only if the "--json" flag is not present.
+	 */
+	displayTable<R extends Ux.Table.Data>(data: R[], columns: Ux.Table.Columns<R>): void;
+
+	/**
+	 * Display an error or message as a warning.
+	 */
+	displayWarning(msg: Error | string): void;
+
+	/**
+	 * Display an error or message as a warning only if the "--verbose" flag is present.
+	 */
+	displayVerboseWarning(msg: Error | string): void;
+
+	/**
+	 * Adds a spinner to the display.
+	 */
+	spinnerStart(msg: string, status?: string): void
+
+	/**
+	 * Updates the status of the spinner in the display.
+	 */
+	spinnerUpdate(status: string): void
+
+	/**
+	 * Appends a heartbeat signal (like a " .") to the existing status of the spinner in the display.
+	 */
+	spinnerWait(): void
+
+	/**
+	 * Stops the spinner in the display.
+	 */
+	spinnerStop(msg: string)
+}
+
+export class UxDisplay implements Display {
+	private readonly displayable: Displayable;
+	private readonly spinner: Spinner;
+	private readonly isVerboseSet: boolean;
+
+	public constructor(displayable: Displayable, spinner: Spinner, isVerboseSet: boolean) {
+		this.displayable = displayable;
+		this.spinner = spinner;
+		this.isVerboseSet = isVerboseSet;
+	}
+
+	public displayInfo(msg: string, ...args: any[]): void {
+		this.displayable.log(msg, ...args);
+	}
+
+	public displayVerboseInfo(msg: string): void {
+		if (this.isVerboseSet) {
+			this.displayInfo(msg);
+		}
+	}
+
+	public async displayConfirmationPrompt(msg: string): Promise<boolean> {
+		return await this.displayable.confirm(msg);
+	}
+
+	public displayStyledHeader(headerText: string): void {
+		this.displayable.styledHeader(headerText);
+	}
+
+	public displayTable<R extends Ux.Table.Data>(data: R[], columns: Ux.Table.Columns<R>): void {
+		this.displayable.table(data, columns);
+	}
+
+	public displayWarning(msg: Error | string): void {
+		this.displayable.warn(msg);
+	}
+
+	public displayVerboseWarning(msg: Error | string): void {
+		if (this.isVerboseSet) {
+			this.displayWarning(msg);
+		}
+	}
+
+	public spinnerStart(msg: string, status?: string): void {
+		this.spinner.start(msg, status);
+	}
+
+	public spinnerUpdate(status: string): void {
+		this.spinner.status = status;
+	}
+
+	public spinnerWait(): void {
+		this.spinner.status += ' .';
+	}
+
+	public spinnerStop(msg: string) {
+		this.spinner.stop(msg);
+	}
+}
+
+
+export interface Displayable {
```

Comment:
You'll notice that our ScannerCommand implements this interface because it is an SfCommand which is a Command. 

Basically these are what we were using before for display purposes to the user. We should not tie ourselves to the fact that SfCommand implements console based output commands for us, which is why this interface is passed into the UxDisplay's constructor... making this implementation detail. Instead we now can solely rely on our own Display interface.